android/pytorch_android/build.gradle

```diff
@@ -79,4 +79,49 @@ task sourcesJar(type: Jar) {
     classifier = 'sources'
 }
 
+def getLibtorchHeadersDir() {
+  def abi = ABI_FILTERS.split(",")[0]
+  return "$rootDir/../build_android_$abi/include";
+}
+
+afterEvaluate {
+  if (POM_PACKAGING == 'aar') {
+    android.libraryVariants.all { variant ->
+      variant.outputs.each { output ->
+        File f = output.outputFile
+        if (f.name.endsWith(".aar")) {
+          output.assemble.finalizedBy addFolderToAarTask(
+              "addHeadersToAar" + variant.name,
+              f.path.replaceAll(variant.name, VERSION_NAME),
+              getLibtorchHeadersDir(),
+              "headers")
```

Comment:
is there any convention about the naming of header directory in AAR?

Reply:
I have not heard about any conventions except [prefab](https://google.github.io/prefab/) that has some structure, but it is not ready for publishing yet. Probably, in 1-2 months we will be able to publish prefab package, but that would be consumable only by android gradle plugin 4.2.0+, gradle 6.5.

In some other packages, like tensorflow lite aar, 'headers' is also used as path for headers inside aar.